pkg/cloud/services.go

```diff
@@ -275,21 +258,81 @@ func (c *AwsServiceClient) GetServiceList(filter *FilterServicesInput) []Service
 	return services
 }
 
-func (c *AwsServiceClient) GetListingFromArn(arn *string) ServiceListing {
-	r, err := regexp.Compile("([^\\/]*)\\/([^-]*)(-.*)?")
+func (c *AwsServiceClient) GetListingFromArn(arn *string) (*ServiceListing, error) {
+	svc, err := c.client.DescribeServices(&ecs.DescribeServicesInput{
+		Cluster: &c.clusterName,
+		Include: []*string{
+			mdl.String(ecs.ServiceFieldTags),
+		},
+		Services: []*string{
+			arn,
+		},
+	})
 
 	if err != nil {
-		panic(err)
+		return nil, err
+	}
+
+	if len(svc.Services) > 1 {
+		return nil, fmt.Errorf("there is more than one service with the specified arn")
```

Comment:
Should we not use errors.New instead (given that there is no string formatting)? Same applies for below

Reply:
fine with that :+1:

Reply:
remember to change the one on line 282 as well

Reply:
nice, did that